pkg/lwip/contrib/sys_arch.c

```diff
@@ -222,4 +224,33 @@ sys_thread_t sys_thread_new(const char *name, lwip_thread_fn thread, void *arg,
     return res;
 }
 
+#ifdef DEVELHELP
+static kernel_pid_t lwip_tcpip_thread = KERNEL_PID_UNDEF;
+void sys_mark_tcpip_thread(void) {
+    lwip_tcpip_thread = thread_getpid();
+}
+
+static kernel_pid_t lwip_lock_thread;
+void sys_lock_tcpip_core(void) {
```

Comment:
Having the variables and functions ordered the following way, makes it a bit more easier to comprehend IMHO.

```suggestion
static kernel_pid_t lwip_tcpip_thread = KERNEL_PID_UNDEF;
static kernel_pid_t lwip_lock_thread;

void sys_mark_tcpip_thread(void) {
    lwip_tcpip_thread = thread_getpid();
}

void sys_lock_tcpip_core(void) {
```

Reply:
Applied the proposed change.

`LWIP_TCPIP_CORE_LOCKING` is on by default when an OS is used and means that other threads than the tcpip thread can access lwIP internals. Without it, messages are sent to the tcpip thread instead where the work is done. This should not be controlled by the DEVELHELP flag.

The new macros here are used to verify that all other threads calling methods actually do take the lock, which seems to match what `DEVELHELP` normally means.

Reply:
`LOCK_TCPIP_CORE` and `UNLOCK_TCPIP_CORE` are only redefined to be able to the the extra thread tracking compared to the standard versions in [tcpip.h](https://git.savannah.nongnu.org/cgit/lwip.git/tree/src/include/lwip/tcpip.h?h=STABLE-2_1_x#n57)